### DISCUSSION TITLE
Speech synthesis: voices are changing when page is hard refreshed

### DISCUSSION THREAD

**Discussion by AstridOttmann** [neuefische/questions#504] [Web]
In my data-object I defined a female voice for speech synthesis. 
When I want to hear the pronunciation, I get first a male voice, only when I refresh the page, I get the female voice. 
For added words I only get the male voice.
What is the reason and how can I change that?

**Comment by doemser** [comment]
Look again at the prototypes from [this discussion](https://github.com/orgs/neuefische/discussions/497).

This prototypes don't have that issue.

It is most likely either the wrong language you are looking for in your voices, or the useEffect mechanics are not yet properly seated.

**Reply by AstridOttmann** [reply]
In the prototypes also one refresh was needed to get the right voice.
Where can I find to choose the voices?

**Reply by doemser** [reply]
This is not understandable for me, this prototype shows the same behavior with you?

https://codesandbox.io/s/speech-synthesis-forked-forked-xe4tws?file=/src/App.js

**Reply by doemser** [reply]
Is this issue solved? Can you please reply?

**Reply by AstridOttmann** [reply]
Not really.
I don't know where to put the hook best, so that the voices are available on the pages I need them. 
So I hard-coded it for now, but I would appreciate a better solution.

**Reply by doemser** [reply]
The hook is best placed in the parent component where you map. If you map in multiple places and need to, you can call the hook in multiple places as well. The only important thing is that you don't call the hook for every list item, because then you call it way too often. 

Another possibility would be to rebuild the hook and store the voices not in a useState but in an atom of jotai and then use the voices globally in your whole app, then it would be enough to call the hook once on the top level.

**Reply by AstridOttmann** [reply]
Thanks! I'll try to rebuild the hook with a global state. Then I would call it in the _app(top level?) ?

But even when it's hard-coded, I don't get the speech synthesis on the smartphone (voice-dropdown is shown, of course, and you can select a voice, but the voice you hear stays always the same)
